ai/agents.md

```
@@ -0,0 +1,195 @@
+
+This is the Kubernetes out-of-tree cloud provider for Azure, implementing the
+[cloud provider interface](https://github.com/kubernetes/cloud-provider). It
+produces three binaries:
+
+- **cloud-controller-manager** — manages Azure resources for a K8s cluster
+- **cloud-node-manager** — runs on each node, manages node-level cloud operations
+- **acr-credential-provider** — provides Azure Container Registry credentials
+
+## Architecture
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)